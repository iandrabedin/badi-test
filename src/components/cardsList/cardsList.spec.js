import React from "react";
import { render, cleanup } from "@testing-library/react";
import CardsList from "./cardsList";

afterEach(cleanup);

const props = {
  recipes: [
    {
      title: "recipe title",
      href: "href",
      ingredients: "potato, cheese",
      thumbnail: "thumbnail.jpg"
    }
  ],
  showEmptyMessage: false
};

describe("CardsList", () => {
  it("should render as expect", () => {
    const { asFragment } = render(<CardsList {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should show empty message if there is no recipe", () => {
    const emptyProps = { ...props, showEmptyMessage: true };
    const { getByText } = render(<CardsList {...emptyProps} />);
    const emptyMessage = getByText(
      "No recipes were found with these ingredients. Try again."
    );
    expect(emptyMessage).toBeInTheDocument();
  });

  it("should show recipes", () => {
    const { getByText } = render(<CardsList {...props} />);
    const recipeTitle = getByText("recipe title");
    expect(recipeTitle).toBeInTheDocument();
  });
});
