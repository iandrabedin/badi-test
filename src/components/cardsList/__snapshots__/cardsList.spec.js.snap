// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList should render as expect 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-sm-12 col-md-6"
    >
      <div
        class="card"
      >
        <div
          class="card__image"
        >
          <a
            href="href"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="recipe title"
              src="thumbnail.jpg"
            />
          </a>
          <div
            class="card__tag-lactose"
          >
            Has Lactose
          </div>
        </div>
        <div
          class="card__info-wrapper"
        >
          <h3
            class="card__title"
          >
            recipe title
          </h3>
          <p
            class="card__description"
          >
            potato, cheese
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
